script.js

```diff
@@ -185,222 +187,240 @@ const books = [
+  const fantasyBooks = books.filter((book)=>book.genre==="Fantasy")
+  printBooks(fantasyBooks)  
+}) 
+
+document.getElementById("adventure").addEventListener("click",()=>{
+  const adventureBooks = books.filter((book)=>book.genre==="Adventure")
+  printBooks(adventureBooks)  
+}) */
+
+
+
+//Search
+userSearch.innerHTML+=`
+  <input type="text" id="userInput" placeholder="  Search your next book">
+  <button id="searchButton">Go</button>
+`
+const searchBooks=()=>{
+  let userInput =""
+  userInput = document.getElementById("userInput")
+  document.getElementById("searchButton").addEventListener("click",(event)=>{
+    const userBookInput = userInput.value.toUpperCase();
+    const searchResult =books.filter((book)=>Object.values(book).join("").toUpperCase().includes(userBookInput));
+    printBooks(searchResult);
+  })
+}
+searchBooks();
+
+
+//Sort
+sort.innerHTML +=`
+  <div class="sortTitle" id="sortTitle" >Sort</div>
+  <div class="sortButton" id="sortButton">
+    <button id="rating" value="rating" >rating ↑↓</button>
+    <button id="year" value="rating">year ↑↓</button>
+    <button id="title" value="title">A-Z ↑↓</button>
+  </div>
+`
+const sortByRating =()=>{
+  let isDescending = false;
+  
+  // Defnine a new array for sorting purpose so books order is not touched. Good explaination here.https://dev.to/samanthaming/how-to-deep-clone-an-array-in-javascript-3cig OR use toSorted() since toSorted will not affect the original array.
+  const ratingBooks = [...books];
+  document.getElementById("rating").addEventListener("click",()=>{
+    console.log(filterSelectedBooks);
+    isDescending = !isDescending;
+    if(isDescending){
+      ratingBooks.sort((a,b)=>(b.rating-a.rating))
+      printBooks(ratingBooks) 
+    } else {
+      ratingBooks.sort((a,b)=>(a.rating-b.rating))
+      printBooks(ratingBooks)  
+    }
+  })
+}
+sortByRating();
+
+const sortByYear=()=>{
+  let isDescending = false;
+  document.getElementById("year").addEventListener("click",()=>{
+    isDescending = !isDescending;
+    if(isDescending){
+      const yearBooks = books.toSorted((a,b)=>(b.year-a.year))
+      printBooks(yearBooks)
+    } else {
+      const yearBooks = books.toSorted((a,b)=>(a.year-b.year))
+      printBooks(yearBooks)
+    }
+  }) 
+}
+sortByYear();
+
+const sortByTitle=()=>{
+  let isAscending = false;
+  document.getElementById("title").addEventListener("click",()=>{
+    isAscending = !isAscending;
+    if(isAscending){
+      const titleBooks = books.toSorted((a,b)=>a.title.localeCompare(b.title))
+      printBooks(titleBooks)
+    } else {
+      const titleBooks = books.toSorted((a,b)=>b.title.localeCompare(a.title))
+      printBooks(titleBooks)
+    }
+  }) 
+}
+sortByTitle();
+
+//Random and filter by century
+surprise.innerHTML +=`
+  <div class="surpriseTitle" id="surpriseTitle">Surprise</div>
+  <div id="surpriseButton">
+    <button id="random" value="random" >A random book</button>
+    <select id="more1">
```

Comment:
The naming here could be more specific. Since you already have a more button you could call it "century-button" instead of "more1". Could also go for the first "more" button for the genre.